utxo/acl_valid_verify.go

```diff
@@ -85,23 +85,27 @@ func (uv *UtxoVM) verifyRWACLValid(tx *pb.Transaction) (bool, error) {
 }
 
 func (uv *UtxoVM) verifyContractValid(tx *pb.Transaction) (bool, error) {
-	req := tx.GetContractRequest()
+	req := tx.GetContractRequests()
 	if req == nil {
 		return true, nil
 	}
 	digestHash, dhErr := txhash.MakeTxDigestHash(tx)
 	if dhErr != nil {
 		return false, dhErr
 	}
-	contractName := req.GetContractName()
-	methodName := req.GetMethodName()
 
-	ok, contractErr := pm.CheckContractMethodPerm(tx.AuthRequire, tx.AuthRequireSigns, digestHash, contractName, methodName, uv.aclMgr)
-	if !ok {
-		uv.xlog.Warn("tx info ", "AuthRequire ", tx.AuthRequire, "AuthRequireSigns ", tx.AuthRequireSigns)
-	}
-	if contractErr != nil {
-		return ok, ErrRWAclNotEnough
+	for i := 0; i < len(req); i++ {
+		tmpReq := req[i]
+		contractName := tmpReq.GetContractName()
+		methodName := tmpReq.GetMethodName()
+
+		ok, contractErr := pm.CheckContractMethodPerm(tx.AuthRequire, tx.AuthRequireSigns, digestHash, contractName, methodName, uv.aclMgr)
+		if !ok {
+			uv.xlog.Warn("tx info ", "AuthRequire ", tx.AuthRequire, "AuthRequireSigns ", tx.AuthRequireSigns)
+		}
+		if contractErr != nil {
+			return ok, ErrRWAclNotEnough
+		}
 	}
-	return ok, nil
+	return true, nil
```

Comment:
You should return the variable of ok instead of true because the function of CheckContractMethodPerm can return a pair of <false, nil>. More details in the file of permission/permission.go